app/memandra.go

```diff
@@ -19,43 +20,35 @@ import (
 	"github.com/spf13/viper"
 )
 
-func init_default_config() {
+func initDefaultConfig() {
 	log.Println("Initializing configuration")
 	viper.SetDefault("ListenPort", 11221)
 	viper.SetDefault("InternalMetricsListenAddress", ":11299")
 	viper.SetDefault("CassandraHostname", "127.0.0.1")
 	viper.SetDefault("CassandraKeyspace", "kvstore")
 	viper.SetDefault("CassandraBucket", "bucket")
-	viper.SetDefault("CassandraBatchBufferItemSize", 80000)
-	viper.SetDefault("CassandraBatchBufferMaxAgeMs", 200*time.Millisecond)
-	viper.SetDefault("CassandraBatchMinItemSize", 1000)
-	viper.SetDefault("CassandraBatchMaxItemSize", 5000)
-	viper.SetDefault("CassandraTimeoutMs", 1000*time.Millisecond)
+	viper.SetDefault("CassandraTimeoutMs", 10000*time.Millisecond)
 	viper.SetDefault("CassandraConnectTimeoutMs", 1000*time.Millisecond)
-}
-
-func load_config_from_env() {
-	log.Println("Mapping configuration from environment")
-	viper.BindEnv("ListenPort", "LISTENPORT")
-	viper.BindEnv("InternalMetricsListenAddress", "METRICSLISTENADDR")
-	viper.BindEnv("CassandraHostname", "CASSANDRAHOST")
-	viper.BindEnv("CassandraKeyspace", "CASSANDRAKEYSPACE")
-	viper.BindEnv("CassandraBucket", "CASSANDRABUCKET")
-	viper.BindEnv("CassandraBatchBufferItemSize", "BUFFERITEMSIZE")
-	viper.BindEnv("CassandraBatchBufferMaxAgeMs", "BUFFERMAXAGE")
-	viper.BindEnv("CassandraBatchMinItemSize", "BATCHMINSIZE")
-	viper.BindEnv("CassandraBatchMaxItemSize", "BATCHMAXSIZE")
-	viper.BindEnv("CassandraTimeoutMs", "CASSANDRATIMEOUT")
-	viper.BindEnv("CassandraConnectTimeoutMs", "CASSANDRACONNTIMEOUT")
+	viper.SetDefault("CassandraNbConcurrentRequests", 500)
+	viper.SetDefault("MemcachedMaxBufferedSetRequests", 80*1000)
+	viper.SetDefault("GracefulShutdownWaitTimeInSec", 10*60)
 }
 
 func main() {
 	if _, set := os.LookupEnv("GOGC"); !set {
 		debug.SetGCPercent(100)
 	}
 
-	init_default_config()
-	load_config_from_env()
+	initDefaultConfig()
+	viper.SetConfigName("config")
+	viper.SetConfigType("yaml")
+	viper.AddConfigPath(".")
+	viper.AddConfigPath("/etc/memandra/")
```

Comment:
I would like to be able to configure that (with a flag)

Reply:
can't manage to get config file and flags at the same time ;(

Reply:
Even by using directly https://golang.org/pkg/flag/ ?